File: proy/conexion/demo/src/main/java/uniandes/edu/co/proyecto/modelo/OrdenDeCompra.java
package uniandes.edu.co.proyecto.modelo;

import java.util.Date;
import java.util.List;

import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import lombok.ToString;

@Document(collection = "orden_de_compra")
@ToString
public class OrdenDeCompra {
    @Id
    private int id;
    private Date fecha_esperada_entrega;
    private String estado;
    private int id_sucursal;
    private int id_proveedor;
    private List<ProductoOrdenCompra> productosOrdenCompra;
}

File: proy/conexion/demo/src/main/java/uniandes/edu/co/proyecto/scripts/insercion/ordenDeCompra.js
db.orden_de_compra.insertOne({
  id: 1,
  fecha_esperada_entrega: ISODate("2024-12-15T00:00:00Z"),
  estado: "pendiente",
  id_sucursal: 2,
  id_proveedor: 10,
  productosOrdenCompra: [
    {
      cantidad: 50,
      precio: 1500,
      id_producto: 101,
    },
    {
      cantidad: 30,
      precio: 2500,
      id_producto: 102,
    },
  ],
});
